extension/intents/navigation/navigation.js

```diff
@@ -9,32 +9,46 @@ import * as browserUtil from "../../browserUtil.js";
 const QUERY_DATABASE_EXPIRATION = 1000 * 60 * 60 * 24 * 30; // 30 days
 const queryDatabase = new Map();
 
+function database(query, tab, url) {
+  queryDatabase.set(query.toLowerCase(), {
+    url,
+    date: Date.now(),
+  });
+  // Sometimes there's a very quick redirect
+  setTimeout(async () => {
+    const newTab = await browser.tabs.get(tab.id);
+    if (newTab.url !== url) {
+      queryDatabase.set(query.toLowerCase(), {
+        url: newTab.url,
+        date: Date.now(),
+      });
+    }
+    saveQueryDatabase();
+  }, 1000);
+}
```

Comment:
There should also be a call to `saveQueryDatabase()` at the end of this function.

Reply:
Sure @ianb I will update this

Reply:
It actually occurred to me later that your code was actually correct, but this code is also correct, so no further updates needed ;)